lib/spack/spack/hooks/sbang.py

```diff
@@ -211,7 +212,7 @@ def install_sbang():
 
     # copy over the fresh copy of `sbang`
     sbang_tmp_path = os.path.join(
-        os.path.dirname(sbang_path), ".%s.tmp" % os.path.basename(sbang_path)
+        os.path.dirname(sbang_path), ".%s.%d.tmp" % (os.path.basename(sbang_path), os.getpid())
```

Comment:
`pid` is not unique across machines or under pid namespaces (docker etc maps them 0)

Reply:
I know, I didn't want to risk regressions with more changes (#33778) for now.

Unless you know of a magic global id, I don't think I'll have time for a better fix right now. If you prefer to wait for a more complete fix, I'll understand, I'll close this PR and keep this fix internally.